SD_Card_Driver/Core/Src/LOS_D_sd.cpp

```diff
@@ -0,0 +1,96 @@
+#include "LOS_D_sd.hpp"
+
+int SDCard::init()
+{
+	return f_mount(&fs, "", 1);
+}
+
+int SDCard::open(const char* location)
+{
+	if (fileOpened && strcmp(location, filePath) == 0) {
+		return 0;
+	}
+
+	if (fileOpened) {
+		int res = f_close(&file);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	int res = f_open(&file, location, FA_OPEN_ALWAYS | FA_WRITE | FA_READ);
+	fileOpened = (res == 0);
+
+	return res;
+}
+
+bool SDCard::checkExist(const char* location)
+{
+	int res = f_open(&file, location, FA_READ);
+	if (res == 0) {
+		f_close(&file);
+	}
+
+	return res == 0;
+}
+
+int SDCard::read(const char* location, uint8_t* data, size_t size, size_t offset)
+{
+	int res = this->open(location);

```

Comment:
I believe the general convention when using the `this->` convention is to only use it when you need to, as in if you have to explicitly say that we want `this` object. So I believe in this case we don't need to specify and it will still work the same